### PR TITLE
Make `oso-host-monitoring` container compatible with CentOS

### PR DESCRIPTION
**Pull request by mhutter**
**Why is this needed**
CentOS does not have RHSM and as such some directories are missing. Without this change the `config.yml` playbook in the oso-host-monitoring container will fail.

**What does it do**
In order to make `src/root/config.yml` a template, I had to change the `generate_containers` role to support Templates in subdirectories.

### CODE REVIEW

**Comment by ivanhorvath** [comment]
You are right, this was done by mistake. We'll check on your PR, we'll have to open it against stg anyway and fix things. Just FYI, we are planning on moving away from this monitoring container, so sooner or later this will be deprecated and there won't be any updates in this code.

File: docker/roles/generate_containers/tasks/main.yml [hunk 1]
```diff
@@ -16,46 +16,32 @@
...
-    path: "{{ item }}"
-  with_fileglob: "{{ base_os }}/*.j2"
-  register: stat_templates
-  when: ansible_version.major > 1
+- name: find templates
+  find:
+    paths: '{{ base_os }}/'
+    patterns: '*.j2'
+    recurse: yes
+  register: find_templates
```
Comment (joelddiaz) [inline docker/roles/generate_containers/tasks/main.yml]: Why not just register this task as 'template_list' and remove the 'set_fact' that follows?

Other than this one thing LGTM.

**Review by joelddiaz — CHANGES_REQUESTED** [review]

File: docker/roles/generate_containers/tasks/main.yml (hunk 1 above)
Comment (mhutter) [reply docker/roles/generate_containers/tasks/main.yml]: Good question... I'll update the PR

Comment (mhutter) [reply docker/roles/generate_containers/tasks/main.yml]: @joelddiaz fixed

**Comment by joelddiaz** [comment]
@mhutter can you open the PR against the 'stg' branch? (leave this one open)

**Comment by mhutter** [comment]
@joelddiaz #2971

**Review by joelddiaz — APPROVED** [review]

**Comment by joelddiaz** [comment]
reviewed #2971 in stg

**Pull request merged**